tests/compatibility-test.py

```diff
@@ -138,38 +147,42 @@ class RayFTTestCase(unittest.TestCase):
 
     @classmethod
     def setUpClass(cls):
-        if not utils.ray_ft_supported(ray_version):
-            raise unittest.SkipTest("ray ft is not supported")
-        utils.delete_cluster()
-        utils.create_cluster()
-        images = [ray_image, kuberay_operator_image, kuberay_apiserver_image]
-        utils.download_images(images)
-        utils.apply_kuberay_resources(images, kuberay_operator_image, kuberay_apiserver_image)
-        utils.create_kuberay_cluster(RayFTTestCase.cluster_template_file,
+        if not utils.is_feature_supported(ray_version, CONST.RAY_FT):
+            raise unittest.SkipTest(f"{CONST.RAY_FT} is not supported")
+        K8S_CLUSTER_MANAGER.delete_kind_cluster()
+        K8S_CLUSTER_MANAGER.create_kind_cluster()
+        image_dict = {
+            CONST.RAY_IMAGE_KEY: ray_image,
+            CONST.OPERATOR_IMAGE_KEY: kuberay_operator_image
+        }
+        operator_manager = OperatorManager(image_dict)
+        operator_manager.prepare_operator()
+        utils.create_ray_cluster(RayFTTestCase.cluster_template_file,
                                      ray_version, ray_image)
 
+    @unittest.skip("Skip test_kill_head due to its flakiness.")
     def test_kill_head(self):
         # This test will delete head node and wait for a new replacement to
         # come up.
-        utils.shell_assert_success(
+        shell_subprocess_run(
             'kubectl delete pod $(kubectl get pods -A | grep -e "-head" | awk "{print \$2}")')
 
         # wait for new head node to start
         time.sleep(80)
-        utils.shell_assert_success('kubectl get pods -A')
+        shell_subprocess_run('kubectl get pods -A')
 
         # make sure the new head is ready
         # shell_assert_success('kubectl wait --for=condition=Ready pod/$(kubectl get pods -A | grep -e "-head" | awk "{print \$2}") --timeout=900s')
         # make sure both head and worker pods are ready
-        rtn = utils.shell_run(
-            'kubectl wait --for=condition=ready pod -l rayCluster=raycluster-compatibility-test --all --timeout=900s')
+        rtn = shell_subprocess_run(
+                'kubectl wait --for=condition=ready pod -l rayCluster=raycluster-compatibility-test --all --timeout=900s')
         if rtn != 0:
-            utils.shell_run('kubectl get pods -A')
-            utils.shell_run(
+            shell_subprocess_run('kubectl get pods -A')
+            shell_subprocess_run(
                 'kubectl describe pod $(kubectl get pods | grep -e "-head" | awk "{print \$1}")')
```

Comment:
Because the default check argument is true in shell_subprocess_run, it will raise an error when the return code of execution is not a 0. Hence, the program will not enter the section below```if rtn != 0``` .

Reply:
Good catch! I will update it. By the way, this test is skipped, and is highly possible to rewrite the test logic.

Reply:
Updated.